src/base.jl

```diff
@@ -2619,6 +2619,28 @@ function convert_component!(
     remove_component!(sys, old_load)
 end
 
+"""
+Set the number of a bus.
+"""
+function set_bus_number!(sys::System, bus::ACBus, number::Int)
+    if number in sys.bus_numbers
+        throw(ArgumentError("bus number $number is already stored in the system"))
+    end
+    bus.number = number
+    push!(sys.bus_numbers, number)
+    return
+end
+
+function set_number!(bus::ACBus, number::Int)
+    Base.depwarn(
```

Comment:
Note that this message is not printed by default. It should show up in tests. I'm debating whether to make it an `@warn` instead.

Reply:
This is ok with me, I don't think this operation is very uncommon anyway.

Reply:
Once we merge to main, I'll merge into `psy5` and delete the deprecated method.